### PR TITLE
WIP: Firms

### PR DESCRIPTION
**Pull request by mappum**
*This PR is a work-in-progress and is not ready to be merged*

This is where I'll be implementing the basic firms functionality as outlined in [this post](https://www.reddit.com/r/MemeEconomy/comments/9prbz4/individual_investment_coin_cap_proposalplan/).

### CODE REVIEW

**Comment by thecsw** [comment]
Hey! Looking forward to seeing this updated!

File: src/main.py [hunk 1]
```diff
@@ -430,5 +452,9 @@ def main():
             traceback.print_exc()
             time.sleep(10)
 
+def concat_names(investors):
+    names = list(map(lambda i: "/u/" + i.name, investors))
```
Comment (thecsw) [inline src/main.py]: Just a little note, MIB is Python3.X oriented, so I would strongly suggest on changing python2 lambda expressions to python3 list comprehension

Comment (thecsw) [reply src/main.py]: I think it would be something like:

``` python
names = ["/u/" + x.name for x in investors]
```

Comment (mappum) [reply src/main.py]: Will do. I'm a python noob so let me know if there is anything else I could do differently.

Comment (thecsw) [reply src/main.py]: I was just taking a quick look, everything looks fantastic! I will leave some comments if I have some. Thank you for your work!

File: src/main.py [hunk 2]
```diff
@@ -278,6 +287,145 @@ def grant(self, sess, comment, grantee, badge):
...
+            first().\
+            name
+        execs = concat_names(
+            sess.query(Investor).\
+                filter(Investor.firm == firm.id).\
+                filter(Investor.firm_role == "exec").\
+                all())
+        traders = concat_names(
+            sess.query(Investor).\
+                filter(Investor.firm == firm.id).\
```
Comment (thecsw) [inline src/main.py:369]: Don't we need to join the tables here?

Comment (mappum) [reply src/main.py:369]: We already did a query to retrieve the firm data above, so it seems like doing the user queries without the join and just using querying by firm id would be a little faster. Or is there a better way of doing it that I'm missing?

**Comment by mappum** [comment]
Closing since all these changes are already in #286

**Pull request closed without merging**